File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/SanPhamBienTheColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SanPhamBienTheRow } from "./SanPhamBienTheRow";

export interface SanPhamBienTheColumns {
    Id: Column<SanPhamBienTheRow>;
    SanPhamTenSp: Column<SanPhamBienTheRow>;
    TenBienThe: Column<SanPhamBienTheRow>;
    GiaBan: Column<SanPhamBienTheRow>;
    GiaKhuyenMai: Column<SanPhamBienTheRow>;
    SoLuong: Column<SanPhamBienTheRow>;
    IsDefault: Column<SanPhamBienTheRow>;
}

export class SanPhamBienTheColumns extends ColumnsBase<SanPhamBienTheRow> {
    static readonly columnsKey = 'SanPhamDB.SanPhamBienThe';
    static readonly Fields = fieldsProxy<SanPhamBienTheColumns>();
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/SanPhamBienThe/SanPhamBienTheColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.SanPhamBienThe")]
[BasedOnRow(typeof(SanPhamBienTheRow), CheckNames = true)]
public class SanPhamBienTheColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string SanPhamTenSp { get; set; }
    [EditLink]
    public string TenBienThe { get; set; }
    public decimal GiaBan { get; set; }
    public decimal GiaKhuyenMai { get; set; }
    public int SoLuong { get; set; }
    public bool IsDefault { get; set; }
}